The default behavior of libressl-portable build system seems to be fairly
broken - it accidentally always enables __STRICT_ALIGNMENT on every arch
(because the BSWAP4 check fails to compile) and even then the BSWAP4 check
appears to be wrong anyway.

Do the same as OpenSSL and always enable the macro except on x86_64, aarch64
and s390(x).

Upstream issue: https://github.com/libressl-portable/portable/issues/602

--- a/configure.ac
+++ b/configure.ac
@@ -75,28 +75,22 @@ AC_ARG_ENABLE([tests],
        [enable_tests="yes"])
 AM_CONDITIONAL([ENABLE_TESTS], [test "x$enable_tests" = xyes])
 
-# Add CPU-specific alignment flags
-old_cflags=$CFLAGS
-CFLAGS="$CFLAGS -I$srcdir/include"
-AC_MSG_CHECKING([if BSWAP4 builds without __STRICT_ALIGNMENT])
-AC_TRY_COMPILE([#include "$srcdir/crypto/modes/modes_lcl.h"],
-	       [int a = 0; BSWAP4(a);],
-	       AC_MSG_RESULT([yes])
-	       BSWAP4=yes,
-	       AC_MSG_RESULT([no])
-	       BSWAP4=no)
-CFLAGS="$old_cflags"
-
 AS_CASE([$host_cpu],
-	[*sparc*], [CPPFLAGS="$CPPFLAGS -D__STRICT_ALIGNMENT"],
 	[*arm*], [host_cpu=arm],
 	[*amd64*], [host_cpu=x86_64 HOSTARCH=intel],
 	[i?86], [HOSTARCH=intel],
 	[x86_64], [HOSTARCH=intel]
 )
-AS_IF([test "x$BSWAP4" = "xyes" -a "$host_cpu" = "arm" ],,CPPFLAGS="$CPPFLAGS -D__STRICT_ALIGNMENT")
 AM_CONDITIONAL([HOST_CPU_IS_INTEL], [test "x$HOSTARCH" = "xintel"])
 
+# Add CPU-specific alignment flags
+AS_CASE([$host_cpu],
+	[*aarch64*], [],
+	[*x86_64*], [],
+	[*s390*], [],
+	[CPPFLAGS="$CPPFLAGS -D__STRICT_ALIGNMENT"]
+)
+
 AC_MSG_CHECKING([if .gnu.warning accepts long strings])
 AC_LINK_IFELSE([AC_LANG_SOURCE([[
 extern void SSLv3_method();
